### DISCUSSION TITLE
Word wrap on left side

### DISCUSSION THREAD

**Discussion by diddiman** [microsoft/monaco-editor#4454] [Q&A]
Hi!

So i am using the monaco diff editor and it works fine. However when i use word wrap i can't seem to get it working for the left side of the side-by-side view. It only wraps on the right side. When the text is too long on the left side it just gives a scrollbar instead. Any idea what it could be?

![Screenshot from 2024-04-01 18-25-59](https://github.com/microsoft/monaco-editor/assets/65303990/f2259975-16f0-4d8c-96b3-bc6a45c82e72)

**Comment by ryexley** [comment]
I'm seeing the same issue as well. Anyone else seen this and have any idea how to fix it?

**Comment by ryexley** [comment]
@diddiman you don't happen to be using the Monaco Editor via the [@monaco-editor/react wrapper](https://github.com/suren-atoyan/monaco-react) do you? I'm asking because I _am_, and I'd like to try to identify whether or not it might have anything to do with that.

**Reply by diddiman** [reply]
I am! I am pretty sure I saw someone using it without the wrapper and it had the same issue. Can't swear on that though

**Reply by ryexley** [reply]
I did _not_ previously have the `monaco-editor` package installed as a direct dependency in my project, so I was depending on whatever version that parent package was pulling in. I think I might possibly the issue solved, but only by adding it as a dependency, and rolling back to run version `0.43.0`, and by doing so, I am no longer seeing the issue. Haven't dug in yet to try to see what might be causing it, but this will do for now for our needs.

**Comment by ryexley** [comment]
From what I can tell, this issue actually exists in VS Code as well. Here's a screenshot of a side-by-side diff where you can fairly clearly see that my code is word-wrapped on the right side, but it is NOT wrapped, and horizontally scrollable on the left.

<img width="1652" alt="image" src="https://github.com/microsoft/monaco-editor/assets/173844/d390ecdf-aab6-4e5e-b19a-0a73f8f15a31">

**Comment by ryexley** [comment]
This is becoming more and more of a mystery to me honestly. For a lack of activity on this conversation, I was going to link to this conversation from a new Bug report in [Issues](https://github.com/microsoft/monaco-editor/issues), but as I was going through the process of doing that, I attempted to create a reproduction of the issue in the [Monaco Editor playground](https://microsoft.github.io/monaco-editor/playground.html), but was unable to do so. [It seems to work for me over there just fine](https://microsoft.github.io/monaco-editor/playground.html?source=v0.47.0#XQAAAAJeCQAAAAAAAABBqQkHQ5NjdMjwa-jY7SIQ9S7DNlzs5W-mwj0fe1ZCDRFc9ws9XQE0SJE1jc2VKxhaLFIw9vEWSxW3yscxGgPALRiQZ_IczC19sQzkLCiiN1oEGBMJVM3p7--pVKVYwzbRy938bXyQdiP7-6I4EYuj_P-ok2wjuea3y5IBidlTmpGRCal4HGYqDhYu01vb07uwN6UyjRMowX1XY9pKMcZE6pbTVoOsE691o7hdqRwaa_QFqyqTngknH6WJcz6zPQMUI35cWtwErlHJGL9gcLhKmr8Ej3O0h6mNaic6E8SpD89mfa4CMKdBTAXjwH3MiPJ0BhE8rxwyfyHnA3KtsS_NzJFKSqaj7RKI6BISZqNtjLJwd3qaKOSKN40fLAdQ9tqJ3rFPj5k69v01dljSLuBe9rR1aCZWckV6IW6H0NXUg7S0OaP1Ynfqe1MwDxop8QdQ4WWR2-3np8MaO3v6uvCw-C_lV7PttyiMarB_l5BXLTCqR6MUre2jVGze-SLTrkNuAYpnaOt3IeBgEEM5asXgAtsBFANZEpMnRwMHYQlSUkrvNlJmcjM5Ao1O9i8zUsRHpXDRiazOQg5QYMpjUyMkv4Eb0eED1PiXfm7_XTT1X3dLIrQOze8eD53mNw7yROG4le_ftYU8aWMyL2FRhyl0HN8KRGbEfsIplpOPSPe03Guywz1ufg8wd6MvMF7XCCmDwYqpq14lUyi2OSW_0nsH-UbhDvWv51ZCSKzQkxtjB4woGzik49ymdTCgghGWu-3vyG76v9KyXq7MarKTOUQfKNeoPtHdysuCjesrdQWTHcnTGczrPdBwt5EW9Zy-i4FR085rdc-d2tgPTAQ-YM3IwCa8zfboDwUHfq08qzn19CmSM6KKoTzkngif5gcaSdAmKnaSR7qeaYZ_28lZB9eOuMTUEVwrdrvzO8gMR4RgWZUyB-YU6CVjXOHFHTYzroNkVCwX4bjPvdKqmmi2aTFK-bAFcZ6CBSco6DVj9XvisyzlMC4bHLXuW1eKNVc1ddLkn41w-eNOC2EiIp9TwZ9dNrh3FdvHGNeUexY6GId5Uri188uQ8mYiWIbU8ZWloeIP2ewmcC3cVyGz2EkENM_hpVXicTBnALJPFAdbolbM-DAI7JF6u3LbNm95GGH24UULV9yKM8MB7mFvcQ7jI7AFo0T7Slz3RlmXvrGtCfE4Jt0gztw9U7ZG0TGv6DW1NcBl92EL3vKQ2pXHGVsMlfSGgtKfk1rOnBN2z6TcBNHIL_ZJtUxPWAXnAF_f7nwpyn_9ht8S8db9uK9l0HfUX6S7wFCkhlIf2p593IXM8i30MIuKpoCjPlnfq7nTTC4uH8UEfJiXDlp7vUEXM5fcUyeWhCEcq_kTcttVVSdYs3vclg8wG4nYvpAit7xMAp8HJ9ScrTA1i0Op1xD_-h30hg). Sssoooo yeah ... I'm not exactly sure what's up with this to be honest.

**Reply by diddiman** [reply]
Hmm, very interesting. That indeed seems to work for me as well...

**Comment by tsx-ell** [comment]
i was facing the exact same issue and setting `options.useInlineViewWhenSpaceIsLimited: false` helped me
```
(property) editor.IDiffEditorBaseOptions.useInlineViewWhenSpaceIsLimited?: boolean | undefined
When renderSideBySide is enabled, useInlineViewWhenSpaceIsLimited is set, and the diff editor has a width less than renderSideBySideInlineBreakpoint, the inline view is used.
```

**Comment by arthurwolf** [comment]
I'm having the exact same issue, in a vue app, no way to get the line wrap to work for the left pane of the diff view.

Anyone has a solution to this? Even like, editing the Monaco source code, **anything**, this is extremely annoying.

Thanks!

**Comment by palbrecht1** [comment]
Having the same issue in 2026

**Reply by Xu-Xihe** [reply]
Using the option `useInlineViewWhenSpaceIsLimited: false` works for me. `React` with `<DiffEditor>` component.

**Comment by Xu-Xihe** [comment]
Find the solution at #4701.
